lib/statsample-glm/glm/formula.rb

```diff
@@ -29,9 +29,11 @@ def parse_formula(form = :token)
 
       private
 
-      def add_contant_term_if_required
-        # TODO: Add support for constants
-        @tokens.unshift Token.new('1')
+      def manage_contant_term
+        @tokens.unshift Token.new('1') unless
+          @tokens.include?(Token.new('1')) ||
+          @tokens.include?(Token.new('0'))
+        @tokens.delete Token.new('0')
```

Comment:
Lets say I implement `Token` as follows:

```ruby
def Token val
  # Since we have to return the same object when val is one. Lets return 0 when values is 1
  return 0 if val == 1
  Token.new(val)
end
```

But now when we will call say `@tokens.sort`, would it also sort Token(1)?
```ruby
@tokens = [Token('a'), Token('a:b'), Token(1)]
@tokens.sort
# It should be [Token(1), Token('a'), Token('a:b')]
```